### PR TITLE
M851 xy 3 point followup

### PR DESCRIPTION
**Pull request by InsanityAutomation**
Followup to #13502 to finish 3 point leveling adjustments. Makes point checks runtime and removes configuration options that no longer make sense.

### CODE REVIEW

**Comment by thinkyhead** [comment]
This needs more work. You can't just use the corners of the mesh bounds as the corners for the triangle when probing the round bed of a delta. That is why there was clever trigonometry to calculate these points for delta. You can't just throw that work away.

File: Marlin/src/inc/SanityCheck.h [hunk 1]
```diff
@@ -1318,12 +1307,9 @@ static_assert(Y_MAX_LENGTH >= Y_BED_SIZE, "Movement bounds (Y_MIN_POS, Y_MAX_POS
...
-    // !! TODO: REPLACE WITH TESTS IN THE Z_SAFE_HOMING CODE !!
-    //
-    //static_assert(WITHIN(Z_SAFE_HOMING_X_POINT, PROBE_X_MIN, PROBE_X_MAX), "Z_SAFE_HOMING_X_POINT is outside the probe region.");
-    //static_assert(WITHIN(Z_SAFE_HOMING_Y_POINT, PROBE_Y_MIN, PROBE_Y_MAX), "Z_SAFE_HOMING_Y_POINT is outside the probe region.");
+  #if HAS_BED_PROBE && (ENABLED(DELTA) || IS_SCARA)
+    static_assert(WITHIN(Z_SAFE_HOMING_X_POINT, PROBE_X_MIN, PROBE_X_MAX), "Z_SAFE_HOMING_X_POINT is outside the probe region.");
+    static_assert(WITHIN(Z_SAFE_HOMING_Y_POINT, PROBE_Y_MIN, PROBE_Y_MAX), "Z_SAFE_HOMING_Y_POINT is outside the probe region.");
   #else
     static_assert(WITHIN(Z_SAFE_HOMING_X_POINT, X_MIN_POS, X_MAX_POS), "Z_SAFE_HOMING_X_POINT can't be reached by the nozzle.");
     static_assert(WITHIN(Z_SAFE_HOMING_Y_POINT, Y_MIN_POS, Y_MAX_POS), "Z_SAFE_HOMING_Y_POINT can't be reached by the nozzle.");
```
Comment (thinkyhead) [inline Marlin/src/inc/SanityCheck.h:1315]: Another consequence that needs to be considered and fulfilled is that all these sanity-checks must be run-time. Please move all these tests where they belong in the code whenever you apply the correction for the 3-point leveling delta coordinates.

File: Marlin/src/inc/Conditionals_post.h [hunk 2]
```diff
@@ -1548,54 +1543,6 @@ constexpr float nozzle_to_probe_offset[XYZ] = NOZZLE_TO_PROBE_OFFSET;
...
-    #endif
-    #ifndef PROBE_PT_2_Y
-      #define PROBE_PT_2_Y (Y_CENTER + (_PROBE_RADIUS) * SIN120)
-    #endif
-    #ifndef PROBE_PT_3_X
-      #define PROBE_PT_3_X (X_CENTER + (_PROBE_RADIUS) * COS240)
-    #endif
-    #ifndef PROBE_PT_3_Y
-      #define PROBE_PT_3_Y (Y_CENTER + (_PROBE_RADIUS) * SIN240)
-    #endif
```
Comment (thinkyhead) [inline Marlin/src/inc/Conditionals_post.h:1576]: Here is the math for delta triangle probe points. Note that only the "probeable radius" is considered — always a circle. This is to ensure centering around 0,0. However, in terms of measuring the tilt of the bed, the "probeable area" is actually an oval when the probe XY offset is non-zero, and the triangle does not need to be equilateral.

**Comment by thinkyhead** [comment]
I'm merging this because it resolves some compile errors. Please follow up quickly with the requested changes and bring Marlin back to correct functionality for deltabots so that we can focus on refactor, documentation, and preparing for release.

**Comment by InsanityAutomation** [comment]
I know but that was as far as I had time to take it. Can't work much past midnight when  I've gotta be up and getting kids to school at 630. It resolved immediate build issues though and was a step better than current so I sent it on up. I'll go after the rest today.

File: Marlin/src/inc/SanityCheck.h (hunk 1 above)
Comment (InsanityAutomation) [reply Marlin/src/inc/SanityCheck.h:1315]: Already is : 
```
#if HOMING_Z_WITH_PROBE
      destination[X_AXIS] -= zprobe_offset[X_AXIS];
      destination[Y_AXIS] -= zprobe_offset[Y_AXIS];
    #endif

    if (position_is_reachable(destination[X_AXIS], destination[Y_AXIS])) {
```

**Pull request merged**